src/language-yaml/utils.js

```diff
@@ -256,7 +256,7 @@ function getBlockValueLineContents(
   const leadingSpaceCount =
     node.indent === null
       ? ((match) => (match ? match[1].length : Number.POSITIVE_INFINITY))(
-          content.match(/^( *)\S/m)
+          content.match(/^( *)[^ \r\n]/m)
```

Comment:
Hm, `\t` was causing the same problem, but it's also fixed by this PR.

current: [Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDJACAPgHShjHGADzwKIE8QAaECABxgEtoBnZUVAJy4gHcACtwTsUqADZ9UFdrQBGXVGADWcGAGVUAWzgAZJlDjIAZhNZx5ilWvX0lBgObIYXAK4WQcLXLgATX366qFAOrqgOcABiEFxaqDDMIchorjAQNCAAFjBa4gDqmUzwrHZgcOoiRUwAbkVUKGCssiAG5lwwAooOcSZmHgBWrMTqjuJwAIquEPC94ua0dlxtyRTa4hn0XAYweUy+MJnIABwADAu85nmK9MmbcG3VRrQAjlPwnQyiaKwAtIZ+fgyXDgryYwM64R6SFMcw85i0TGcbjhowmbyM0L6tBgqDku32hyQACZsYomOJHABhCBaKGeVgAVgyrnMABVcaIYfMQNV3ABJKABWDqMBbRgAQUF6hgFDGs3MAF8FUA)
expected: no changes ([Yaml online parser link](
https://yaml-online-parser.appspot.com/?yaml=a%3A+%7C%0A++%09x%0A++%09y&type=json))

and it behaves as expected in this PR's change.

Reply:
Are the the html space? https://github.com/prettier/prettier/blob/8c49fd0f6f49aca95e5004ff96f5c9fe9f95f3f8/src/language-html/utils/index.js#L24

Reply:
@fisker I don't think so.
For example, if you modify this regex to `/^( *)[^ \t\r\n]/m`, [the case I showed above](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDJACAPgHShjHGADzwKIE8QAaECABxgEtoBnZUVAJy4gHcACtwTsUqADZ9UFdrQBGXVGADWcGAGVUAWzgAZJlDjIAZhNZx5ilWvX0lBgObIYXAK4WQcLXLgATX366qFAOrqgOcABiEFxaqDDMIchorjAQNCAAFjBa4gDqmUzwrHZgcOoiRUwAbkVUKGCssiAG5lwwAooOcSZmHgBWrMTqjuJwAIquEPC94ua0dlxtyRTa4hn0XAYweUy+MJnIABwADAu85nmK9MmbcG3VRrQAjlPwnQyiaKwAtIZ+fgyXDgryYwM64R6SFMcw85i0TGcbjhowmbyM0L6tBgqDku32hyQACZsYomOJHABhCBaKGeVgAVgyrnMABVcaIYfMQNV3ABJKABWDqMBbRgAQUF6hgFDGs3MAF8FUA) cannot be correctly treated.

This `leadingSpaceCount` variable should be calculated as `Infinity` for blank line. The problem is, for string values such like `  \tx` or `  \u3000x`, `leadingSpaceCount` becomes `Infinity` (it should be `2`) because `\S` includes `\t`, `\u3000` and other characters which shouldn't be treated as blank.

I'm not expert of Yaml so actually I don't know this change solves all problems, but I believe this change just improve code quality. `\S` seems just mistakenly chosen.

Reply:
Looks like the existing code does not account for `\t` being used for indents (at least var name – `leadingSpaceCount` – suggests that). Might be worth considering given https://github.com/prettier/prettier/issues/7475

Reply:
@kachkaev YAML does not allow tabs as indents. https://yaml.org/spec/1.2.2/#61-indentation-spaces

Reply:
I mean, `\t`, `\u3000` (and some other characters included in `\S`) after indentation spaces should be treated just as normal characters. `\r` and `\n` are special because a line which includes only {space, `\r`, `\n`} should be treated as empty line (i.e. leadingSpaceCount is Infinity) so `\r` and `\n` are specified in this PR's change.

Reply:
Should we consider U+2028 and U+2029?

Reply:
I guess it's answered here https://yaml.org/spec/1.2.2/#54-line-break-characters